spec/features/admin/accounting/transaction_flow_spec.rb

```diff
@@ -30,54 +30,49 @@
     end
 
-    # TODO: Figure out whether this makes sense as a test case anymore
-    xdescribe 'transaction listing' do
-      scenario 'with qb error during Updater' do
-        Rails.configuration.x.test.raise_qb_error_during_updater = 'qb fail on index'
-        visit "/admin/loans/#{loan.id}/transactions"
-        expect(page).to have_alert('Some data may be out of date. (Error: qb fail on index)')
-      end
-    end
+    describe 'list transactions' do
+      context 'when transactions present' do
+        let!(:transactions) { create_list(:accounting_transaction, 2, project: loan, description: "Pants") }
 
-    context "loan's transactions are read-only" do
-      let!(:loan) { create(:loan, txn_handling_mode: Loan::TXN_MODE_READ_ONLY, division: division) }
-      scenario 'create new transaction button is hidden' do
-        visit "/admin/loans/#{loan.id}/transactions"
-        expect(page).to have_content('Transactions are read-only')
-        # expect "Add Transaction" to not be available
-        expect(page).not_to have_selector('.btn[data-action="new-transaction"]')
-      end
-    end
+        scenario 'shows transactions' do
+          visit "/admin/loans/#{loan.id}/transactions"
+          amt = ActiveSupport::NumberHelper.number_to_delimited(transactions[0].amount)
+          expect(page).to have_content(amt)
+          expect(page).to have_content("Pants")
+        end
 
-    context "loan's qb_division has qb_read-only on" do
-      before do
-        Division.root.update_attribute(:qb_read_only, true)
-      end
-      scenario 'create new transaction button is hidden' do
-        visit "/admin/loans/#{loan.id}/transactions"
-        # expect "Add Transaction" to not be available
-        expect(page).not_to have_selector('.btn[data-action="new-transaction"]')
-      end
-    end
+        context 'when transactions are writable' do
+          scenario 'create new transaction button is visble, no notice shown' do
+            visit "/admin/loans/#{loan.id}/transactions"
+            expect(page).not_to have_content("You can't add transactions")
+            expect(page).to have_selector('.btn[data-action="new-transaction"]')
+          end
+        end
 
-    context "loan is not active" do
-      let!(:loan) { create(:loan, :completed, division: division) }
-      scenario 'create new transaction button is hidden' do
-        visit "/admin/loans/#{loan.id}/transactions"
-        expect(page).to have_content('Transactions are read-only')
-        # expect "Add Transaction" to not be available
-        expect(page).not_to have_selector('.btn[data-action="new-transaction"]')
+        context "when transactions are not writable" do
+          let!(:loan) { create(:loan, :completed, division: division) }
+
+          before do
+            Division.root.update_attribute(:qb_read_only, true)
+          end
+
+          scenario 'create new transaction button is hidden and reasons are shown' do
```

Comment:
I took out a bunch of other tests for different reasons since that stuff is now covered in policy spec. Moving it down the testing pyramid!